### PR TITLE
Update docs (multiple minor fixes)

### PR DESCRIPTION
**Pull request by mloning**
#### Reference Issues/PRs
Related to #1159 

#### What does this implement/fix? Explain your changes.
* move file content to rst files in docs folder, keeping md files with links to website in root dir
* converts file in root dir to markdown (rst doesn't render correctly)
* Key documents are copied over and unchanged (governance, code of conduct)
* fixes broken cross-references in docs (contributing)
* add roles to website (based on PR #1296)
* uses sphinx extension `github_issues` to link to GitHub issues, PRs and users
* move STEP description to website, combining info from governance document and repository
* fixes sphinx warning by fixing missing data set export in init file for `load_acsf1`

### CODE REVIEW

**Comment by mloning** [comment]
@RNKuhns could you review? It's a lot of files, but most changes should be straightforward. 

@fkiraly and @TonyBagnall you may also want to see if the changes are okay with you since it moves the governance document and code of conduct files as described above.

**Comment by RNKuhns** [comment]
@mloning I've started to go through this. 

General question. When should we use the \``sktime\`` highlighting in the website text. On several pages we use it sometimes but not others (I'm not really sure when we should or shouldn't). 

I'll try to keep a running total of things that need tweaks below:

- Index
    - Users panel top heading should say "Documentation" so it matches the link heading at the top of the page
    - Developers panel top heading should say "Development" so it matches the link heading at the top of the page  
- About
    - Contributors panel links to governance contributors section, not the all contributors
    - Citing sktime panel link doesn't work

File: docs/source/get_involved/contributing.rst
Comment (RNKuhns) [inline docs/source/get_involved/contributing.rst:362]: I'd add on to the sentence : "along with sktime specific conventions described in our :ref:`developer guide's documentation section <developer_guide_documentation>`

**Comment by mloning** [comment]
Thanks @RNKuhns 

> When should we use the ``sktime`` highlighting in the website text. On several pages we use it sometimes but not others (I'm not really sure when we should or shouldn't).

I don't have a strong preference, but if I can choose I would not use the highlighting and reserve the highlighting for code snippets, mostly because highlighting the name of sktime doesn't really help readability.

> Do we want the new markdown documents in the repo root directory to reference the latest docs or the most recent release? I'm thinking we have to do latest now so there is the document to reference, but not sure what the right answer is for later; probably latest too.

Yes, I thought that for these documents (perhaps except for the change log) it's best to point to the latest versions, as the changes to these documents take effect as soon as they are merged into `main` and they don't depend on code releases.

File: docs/source/get_involved/contributing.rst
Comment (mloning) [reply docs/source/get_involved/contributing.rst]: Added the sentence

**Review by fkiraly — CHANGES_REQUESTED** [review]
I see you added a "roles" page, as an alternative suggestion to my PR in #1296.

Don't mind about the formatting, but:

* I think your list of core contributors is inaccurate, it's missing people and/or incorrectly lists them as inactive? Compare #1296
* The contributors are in no particular order. I would do last name, first name, and then order by last name
* the list of names is not very findable. It should be directly linked from the landing page in an "obvious" location, e.g., a "team" or "developers" rubric.

**Comment by mloning** [comment]
Thanks for the suggestions @fkiraly. I added more cross-references, a link to the roles page in the readme and ordered the contributors by first name (more personal than going by last name). 

We don't have any precise rules on inactivity of core devs as far as I'm aware, but I think the list makes sense now as it is.

File: docs/source/about/history.rst
Comment (fkiraly) [inline docs/source/about/history.rst:18]: you are

**Review by fkiraly — DISMISSED** [review]
My points about the roles were addressed, so I'm happy to merge this (under the assumption that only the location changes, not content).

File: docs/source/about/history.rst
Comment (mloning) [inline docs/source/about/history.rst:18]:
```suggestion
If you are interested in contributing, check out our :ref:`get_involved` guide.
```

**Pull request merged**
